### PR TITLE
Update Models To Typescript

### PR DESCRIPTION
**Pull request by ericyliu**
Links
-----
- Issue: _link to GitHub issue (or remove this line)_
- Web repo PR: _link to companion PR in server repo (or remove this line)_
- Dependency PRs: _links to PRs that must be merged prior this one (or remove this line)_

Description
-----------
- updates some models to typescript (didn't do all because this pr was already becoming quite big)

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

File: models/feedback.ts [hunk 1]
```diff
@@ -1,6 +1,17 @@
-var mongoose = require('mongoose')
-
-var feedbackSchema = new mongoose.Schema({
+import { Document, model, Schema } from 'mongoose';
+
+export interface Feedback extends Document {
+  sessionId: string;
+  type: string;
+  subTopic: string;
+  responseData: { [key: string]: any };
```
Comment (ericyliu) [inline models/feedback.ts:7]: this is what sendgrid expects (as far as I can tell, this is being used to send stuff to sendgrid)

File: models/feedback.ts [hunk 2]
```diff
@@ -1,6 +1,17 @@
-var mongoose = require('mongoose')
-
-var feedbackSchema = new mongoose.Schema({
+import { Document, model, Schema } from 'mongoose';
+
+export interface Feedback extends Document {
+  sessionId: string;
+  type: string;
```
Comment (ericyliu) [inline models/feedback.ts:5]: this might need to be more restrictive than string

File: models/ipAddress.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Document, model, Schema, Types } from 'mongoose';
+import { IP_ADDRESS_STATUS } from '../constants';
+
+export interface IpAddress extends Document {
+  createdAt: Date;
+  ip: string;
+  users: Types.ObjectId[];
+  status: IP_ADDRESS_STATUS.OK | IP_ADDRESS_STATUS.BANNED;
```
Comment (ericyliu) [inline models/ipAddress.ts:8]: maybe consider moving `IP_ADDRESS_STATUS` to this model file?

**Comment by ericyliu** [comment]
changed model file names from uppercase start to lowercase start, to make it consistent with the rest of the project, i can revert this if we feel strongly about this change

File: models/message.ts [hunk 4]
```diff
@@ -25,25 +27,25 @@ var messageSchema = new mongoose.Schema(
...
 
 messageSchema.virtual('picture').get(function() {
   // only works if user is populated
-  return this.user.picture
-})
+  return this.user.picture;
+});
 
-module.exports = mongoose.model('Message', messageSchema)
+export default model<Message>('Message', messageSchema);
```
Comment (ericyliu) [inline models/message.ts:51]: not sure if I should include virtuals in the typing (i think i will need to later, when the files using the model is converted and its expecting those keys to be there)

File: models/feedback.ts (hunk 1 above)
Comment (jkeat) [reply models/feedback.ts:7]: hmm this isn't related to sendgrid
https://github.com/UPchieve/server/blob/master/router/api/feedback.js#L10
https://github.com/UPchieve/web/blob/master/src/views/FeedbackView.vue#L334

Comment (ericyliu) [reply models/feedback.ts:7]: err the only way its used in the api at least is passed into SendGrid through MailService
https://github.com/UPchieve/server/blob/master/services/MailService.js#L67

**Pull request merged**